mulog-kinesis/src/com/brunobonacci/mulog/publishers/kinesis.clj

```diff
@@ -0,0 +1,71 @@
+(ns com.brunobonacci.mulog.publishers.kinesis
+  (:require [com.brunobonacci.mulog.buffer :as rb]
+            [com.brunobonacci.mulog.utils :as ut]
+            [com.brunobonacci.mulog.publishers.aws-utils :as awsutils]
+            [cheshire.core :as json]
+            [cheshire.generate :as gen]))
+
+;;
+;; Add Flake encoder to JSON generator
+;;
+(gen/add-encoder com.brunobonacci.mulog.core.Flake
+                 (fn [x ^com.fasterxml.jackson.core.JsonGenerator json]
+                   (gen/write-string json ^String (str x))))
+
+(defn- put-records
+  [{:keys [stream-name partition-key-name format max-items] :as config} records]
+  (let [key-field partition-key-name
+        fmt* (if (= :json format) json/generate-string ut/edn-str)]
+    (->> records
+         (map (juxt #(str (get % key-field)) fmt*))
+         (map (fn [[k v]] (awsutils/create-record! k v)))
+         (partition-all max-items)
+         (map #(awsutils/publish! stream-name %))
+         (doall))))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;                                                                                ;;
+;;                         ----==| K I N E S I S |==----                          ;;
+;;                                                                                ;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(deftype KinesisPublisher
+  [config buffer transform]
+
+  com.brunobonacci.mulog.publisher.PPublisher
+  (agent-buffer [_]
+    buffer)
+
+  (publish-delay [_]
+    (:publish-delay config))
+
+  (publish [_ buffer]
+    ;; items are pairs [offset <item>]
+    (let [items (take (:max-items config) (rb/items buffer))
```

Comment:
here we should support the `transform` as well 

see: https://github.com/BrunoBonacci/mulog/blob/master/doc/custom-publishers.md#support-for-user-defined-transformations

Reply:
I pass `transform` via constructor
https://github.com/BrunoBonacci/mulog/pull/21/files#diff-275c60150df4c95c632cfc31584a2af2R33
https://github.com/BrunoBonacci/mulog/pull/21/files#diff-275c60150df4c95c632cfc31584a2af2R71
Do you find it ok or should I rework it?